### PR TITLE
[MPS] aten::erfinv ops

### PR DESCRIPTION
**Pull request by TaiPhamD**
I've added the implementation of erfinv using the algorithm from https://github.com/pytorch/pytorch/blob/4154c8ea159fdaecc71ee9af820ac956193c875b/aten/src/ATen/native/Math.h#L152 in order for the MPS based algorithm to match the CPU automatic test.

Testing shows MPS has a decent speed up compared to CPU. 

```python
import torch
x = torch.arange(-1, 1, 1e-8) # default cpu tensor
#measure CPU compute time by calling torch.erfinv
time = %timeit -o -q torch.erfinv(x)
cpu_time = time.average
print("CPU torch.erfinv time: ", cpu_time)

x = x.to("mps")
# measure MPS compute time
time = %timeit -o -q  torch.erfinv(x)
mps_time = time.average
print("MPS torch.erfinv time: ", mps_time)
print(f"MPS torch.erfinv is {cpu_time/mps_time*100} percent faster than CPU torch.erfinv")

# compute MSE between MPS and CPU torch.erfinv
x = x.to("cpu")
print("Done convert x to cpu device")
y_cpu = torch.erfinv(x)
print("Done computing cpu torch.erfinv")
x = x.to("mps")
print("Done convert x to mps device")
y_mps = torch.erfinv(x)
print("Done computing mps torch.erfinv")

y_mps = y_mps.to("cpu")
print("computing mask")
mask = torch.isfinite(y_cpu) & torch.isfinite(y_mps.to("cpu"))
y_mps = y_mps[mask]
y_cpu = y_cpu[mask]
x = x[mask]
print
mse = torch.square(y_cpu - y_mps).mean()
print("MSE between MPS and CPU torch.erfinv: ", mse)
diff = torch.abs(y_cpu - y_mps)

print("Largest difference")
print(f"x:  {x[torch.argmax(diff)]}, y_cpu: {y_cpu[torch.argmax(diff)]}, y_mps: {y_mps[torch.argmax(diff)]} , diff = {y_cpu[torch.argmax(diff)] - y_mps[torch.argmax(diff)]}")
```
CPU torch.erfinv time:  2.6942295774279046
MPS torch.erfinv time:  0.1837057119286328
MPS torch.erfinv is 1466.600874378134 percent faster than CPU torch.erfinv
Done convert x to cpu device
Done computing cpu torch.erfinv
Done convert x to mps device
Done computing mps torch.erfinv
computing mask
MSE between MPS and CPU torch.erfinv:  tensor(4.1189e-14)
Largest difference
x:  -0.6999978423118591, y_cpu: -0.7328658699989319, y_mps: -0.7328646779060364 , diff = -1.1920928955078125e-06

Fixes #https://github.com/pytorch/pytorch/issues/86808

### CODE REVIEW

**Comment by TaiPhamD** [comment]
mps test failed will investigate.

**Comment by TaiPhamD** [comment]
well it passed the baby uni op test that I added but failed on this currently:

```
❯ python3 test/test_mps.py TestConsistencyCPU.test_output_grad_match_erfinv_cpu_float32
F
======================================================================
FAIL: test_output_grad_match_erfinv_cpu_float32 (__main__.TestConsistencyCPU)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/rush/src/pytorch/torch/testing/_internal/common_device_type.py", line 401, in instantiated_test
    result = test(self, **param_kwargs)
  File "/Users/rush/src/pytorch/torch/testing/_internal/common_device_type.py", line 846, in test_wrapper
    return test(*args, **kwargs)
  File "/Users/rush/src/pytorch/test/test_mps.py", line 10492, in test_output_grad_match
    self.assertEqual(cpu_grad_inputs, mps_grad_inputs, atol=atol, rtol=rtol)
  File "/Users/rush/src/pytorch/torch/testing/_internal/common_utils.py", line 3048, in assertEqual
    raise error_metas[0].to_error(
AssertionError: Tensor-likes are not close!

Mismatched elements: 3 / 400 (0.8%)
Greatest absolute difference: 0.005126953125 at index (12, 3) (up to 0.0001 allowed)
Greatest relative difference: 4.336770871304907e-05 at index (12, 3) (up to 1.3e-06 allowed)

The failure occurred for item [0]

----------------------------------------------------------------------
Ran 1 test in 0.144s
```

- Perhaps more rounds of newton methods needed
- or  just have to redesign the compute graph to follow more closely with the CPU approximation method than using the logarithm based approach.

File: aten/src/ATen/native/mps/operations/UnaryOps.mm [hunk 1]
```diff
@@ -250,6 +251,93 @@ void unary_op(const Tensor& self,
   });
 }
 
+TORCH_IMPL_FUNC(erfinv_out_mps)(const Tensor& self, const Tensor& output) {
+  mps::unary_op(self, output, "erfinv_out_mps", ^MPSGraphTensor*(MPSGraph* mpsGraph, MPSGraphTensor* inputTensor) {
+    auto dataType = inputTensor.dataType;
```
Comment (kulinseth) [inline aten/src/ATen/native/mps/operations/UnaryOps.mm:256]: Can you please provide a link or in comment provide formula you are implementing for this op?

**Comment by TaiPhamD** [comment]
curent CI failure is due to:

2023-05-11T08:23:21.1161180Z ==================================== RERUNS ====================================
2023-05-11T08:23:21.1161290Z _______ TestConsistencyCPU.test_output_match_grid_sampler_2d_cpu_float32 _______
2023-05-11T08:23:21.1161330Z Unexpected success
2023-05-11T08:23:21.1161430Z _______ TestConsistencyCPU.test_output_match_grid_sampler_2d_cpu_float32 _______
2023-05-11T08:23:21.1161470Z Unexpected success
2023-05-11T08:23:21.1161540Z =================================== FAILURES ===================================
2023-05-11T08:23:21.1161640Z _______ TestConsistencyCPU.test_output_match_grid_sampler_2d_cpu_float32 _______
2023-05-11T08:23:21.1161680Z Unexpected success
2023-05-11T08:23:21.1161980Z - generated xml file: /Users/sk8/actions-runner/_work/pytorch/pytorch/test/test-reports/python-pytest/test_mps/test_mps-790bccb3bae8d6eb.xml -
```

it has 1 extra unexpected success. Is it because of my newly added erfinv ?

**Comment by TaiPhamD** [comment]
closing this one in favor of: https://github.com/pytorch/pytorch/pull/101507/files

raw metal kernel is 18x faster

**Pull request closed without merging**